src/peer_connection.cpp

```diff
@@ -2124,7 +2124,7 @@ namespace libtorrent
 		if (t->share_mode()) return false;
 
 		if (m_upload_only && t->is_upload_only()
-			&& can_disconnect(error_code(errors::upload_upload_connection, get_libtorrent_category())))
+			&& can_disconnect(errors::upload_upload_connection))
```

Comment:
yeah, I think so. previously I was a bit confused about error_code and error_condition, and I had marked the libtorrent error codes as both, which made this not automatically pick the right constructor..